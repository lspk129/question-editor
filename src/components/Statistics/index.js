import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import getStatistics from '../../utils';

const Section = styled.div`
  padding: 0 20px 20px;
  ${({ stats }) => stats && `
    padding-top: 50px;
  `};
`;

const Statistics = ({ state }) => {
  const {
    rowNumber,
    colNumber,
    imgNumber,
    longestRowLabel,
    longestColLabel,
  } = getStatistics(state);

  return (
    <Section>
      <h2>Question Summary View</h2>
      <h3>Summary</h3>
      <Section stats>
        <p>Number of rows: {rowNumber}</p>
        <p>Number of columns: {colNumber}</p>
        <p>Number of images uploaded: {imgNumber}</p>
        <p>Longest row label: {longestRowLabel}</p>
        <p>Longest column label: {longestColLabel}</p>
      </Section>
    </Section>
  );
};

Statistics.propTypes = {
  state: PropTypes.shape({
    rows: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    })).isRequired,
  }).isRequired,
};

export default Statistics;
