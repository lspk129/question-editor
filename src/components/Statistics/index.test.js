import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Statistics from './';

describe('Statistics component', () => {
  const props = {
    rows: [
      { id: '1', text: 'hello', img: [] },
    ],
    cols: [
      { id: '2', text: '', img: ['1'] },
      { id: '3', text: 'world', img: [] },
    ],
  };
  const wrapper = mount(<Statistics state={props} />);

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has state prop', () => {
    expect(wrapper).toHaveProp('state');
  });

  it('has h2 title with copy "Question Summary View"', () => {
    expect(wrapper.find('h2')).toHaveText('Question Summary View');
  });

  it('has copy "Number of rows"', () => {
    expect(wrapper.find('p').at(0)).toHaveText('Number of rows: 1');
  });

  it('has copy "Number of columns"', () => {
    expect(wrapper.find('p').at(1)).toHaveText('Number of columns: 2');
  });

  it('has copy "Number of images uploaded"', () => {
    expect(wrapper.find('p').at(2)).toHaveText('Number of images uploaded: 1');
  });

  it('has copy "Longest row label"', () => {
    expect(wrapper.find('p').at(3)).toHaveText('Longest row label: 5');
  });

  it('has copy "Longest column label"', () => {
    expect(wrapper.find('p').at(4)).toHaveText('Longest column label: 5');
  });
});
