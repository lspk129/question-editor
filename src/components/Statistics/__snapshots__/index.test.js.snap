// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics component renders correctly 1`] = `
<Statistics
  state={
    Object {
      "cols": Array [
        Object {
          "id": "2",
          "img": Array [
            "1",
          ],
          "text": "",
        },
        Object {
          "id": "3",
          "img": Array [],
          "text": "world",
        },
      ],
      "rows": Array [
        Object {
          "id": "1",
          "img": Array [],
          "text": "hello",
        },
      ],
    }
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa floIIC"
    >
      <h2>
        Question Summary View
      </h2>
      <h3>
        Summary
      </h3>
      <styled.div
        stats={true}
      >
        <div
          className="sc-bdVaJa hMXEoI"
        >
          <p>
            Number of rows: 
            1
          </p>
          <p>
            Number of columns: 
            2
          </p>
          <p>
            Number of images uploaded: 
            1
          </p>
          <p>
            Longest row label: 
            5
          </p>
          <p>
            Longest column label: 
            5
          </p>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Statistics>
`;
